src/imp/schannel.rs

```diff
@@ -294,6 +315,24 @@ impl<S: io::Read + io::Write> TlsStream<S> {
         }
     }
 
+    pub fn certificate_chain(&mut self) -> Result<ChainIterator<S>, Error> {
+        if self.1.is_none() {
+            match self.0.peer_certificate() {
+                Ok(cert) => {
+                    self.1 = cert.cert_store();
```

Comment:
Why do we need to cache the cert store?

Reply:
I could not find a better solution. CertStore once you call certs is borrowed. You can't drop it and you can't put it in the iterator as it will be a self referenced struct.

Reply:
It seems like things would be easier to work with if we just returned a `Result<Vec<Certificate>>`. I don't think we're gaining all that much by not buffing the certs up again.